tests/impl.cpp

```diff
@@ -1525,6 +1576,114 @@ bool test_mm_cmplt_ps(const float *_a, const float *_b)
     return validateInt32(iret, result[0], result[1], result[2], result[3]);
 }
 
+bool test_mm_cmplt_ss(const float *_a, const float *_b)
+{
+    __m128 a = test_mm_load_ps(_a);
+    __m128 b = test_mm_load_ps(_b);
+
+    float result[4];
+    result[0] = _a[0] < _b[0] ? getNAN() : 0;
```

Comment:
@jserv do you think `getNAN()` would be a little confusing?

Reply:
The function `getNAN` was intended to check the propagation of quiet/signal NaN through arithmetic operations. Thus, `get` might be confusing.

Reply:
I think @afcidk use it in this way is probably caused by referencing the other source code such as `compord()`, which returns `getNAN()`.
We might need to replace these confusing `getNAN()` in the future,

Reply:
I need a number to represent floating number whose bits are all one (0b11111......), and saw that the `getNAN()` function actually use the number I expected. NaN representation only guarantees the exponent part to be all 1 and non-zero number in mantissa.

I think using `getNaN()` is confusing, should I introduce another function?

Reply:
How about defining the macros like
```c
#define SET_ALL_BIT_1_64 0x11...11
#define SET_ALL_BIT_1_32 0x1..1
```

Reply:
Something like `~0`?

Reply:
`~0` sounds good to me.

Reply:
We cannot assign `~0` to floating point number directly, that would be represented as -1.000 instead. I like the original implementation of `getNAN()` to cast int32_t to float. Defining it as a macro sounds good, but it seems like we still need an integer-typed variable?
```
const int32_t nan = ~0;
const int64_t nan_64 = ~0;
#define SET_ALL_BIT_1_32 *(float*)&nan;
#define SET_ALL_BIT_1_64 *(double*)&nan_64;
```

Reply:
The return value is clearer in NEON ISA than in SSE.
vcleq_f32 takes floating point numbers and returns uint32_x4_t (for which 0xFFFFFFFF makes sense), whereas _mm_cmple_ps returns the same value, but in a __m128 instead of a __m128i.

It think it would be less confusing to code the way you did with test_mm_cmpneq_ps, than to use getNAN :
`bool test_mm_cmpneq_ps(const float *_a, const float *_b)
{
    __m128 a = test_mm_load_ps(_a);
    __m128 b = test_mm_load_ps(_b);

    int32_t result[4];
    result[0] = _a[0] != _b[0] ? -1 : 0;
    result[1] = _a[1] != _b[1] ? -1 : 0;
    result[2] = _a[2] != _b[2] ? -1 : 0;
    result[3] = _a[3] != _b[3] ? -1 : 0;

    __m128 ret = _mm_cmpneq_ps(a, b);
    __m128i iret = *(const __m128i *) &ret;
    return validateInt32(iret, result[0], result[1], result[2], result[3]);
}`

Reply:
I think using macros like `SET_ALL_BIT_1_32` should be easy to understand as well once we give up the `getNAN` usage in this `0xFFFFFFFF` value.

I feel fine to both implementation, how do you think @marktwtn ?

Reply:
I choose the macro implementation.